File: 94. Binary Tree Inorder Traversal.py
class Solution:
    def inorderTraversal(self, root):

        result = []

        def inorder(node):

            # Base case
            if not node:
                return

            # Traverse left subtree
            inorder(node.left)

            # Visit current node
            result.append(node.val)

            # Traverse right subtree
            inorder(node.right)

        inorder(root)

        return result

File: 94. Binary Tree Inorder Traversal.java
class Solution {

    private List<Integer> result = new ArrayList<>();

    private void inorder(TreeNode root) {

        // Base case
        if (root == null) {
            return;
        }

        // Traverse left subtree
        inorder(root.left);

        // Visit current node
        result.add(root.val);

        // Traverse right subtree
        inorder(root.right);
    }

    public List<Integer> inorderTraversal(TreeNode root) {

        inorder(root);

        return result;
    }
}
